src/generic/datavgen.cpp

```diff
@@ -2176,12 +2226,18 @@ wxDragResult wxDataViewMainWindow::OnData( wxDataFormat format, wxCoord x, wxCoo
     if ( row < GetRowCount() && xx <= GetEndOfLastCol() )
         item = GetItemByRow( row );
 
-    wxCustomDataObject *obj = (wxCustomDataObject *) GetDropTarget()->GetDataObject();
+    wxDataObjectComposite *obj = static_cast<wxDataObjectComposite*>(GetDropTarget()->GetDataObject());
+
+    size_t dataSize = obj->GetDataSize(format);
+    wxMemoryBuffer buffer;
+    obj->GetDataHere(format, buffer.GetWriteBuf(dataSize));
+    buffer.UngetWriteBuf(dataSize);
```

Comment:
Data could be relatively big here, I wonder if it could be worth it to retrieve it only on demand from `wxDataViewEvent::GetDataBuffer()` instead of doing it like this.

Reply:
Ok, I can update `wxDataViewEvent::GetDataBuffer()` method.